### ISSUE TITLE
Building with MSVC on windows

### ISSUE THREAD

**Issue by Talhada** [frno7/libpes#4]
When building the library with mscv on windows (Visual Studio 2022), I get the following error:

>\libpes\src\sax.c(370): error C2059: syntax error: '}'

As far as I'm concerned about C code, this isn't a C standard and rather a C++ standard for object initialization. To make the project 100% cross platform I would recommend to initialize C structs manually (there are not many of them)

### Example:
````diff
-struct sax_state state = {};
+struct sax_state state;
+state.level = 0;
````

**Comment by frno7** [comment]
Hi @Talhada, and thanks for the report. Well, it’s a GCC and Clang extension which is quite good, because if a new field is added to the `sax_state` structure, your code won’t zero it, and it’s likely to be uninitialised, which would be dangerous. Instead, one could do `memset(&state, 0, sizeof(state));` to be safe, but that’s rather awkward. Sometimes it’s done like `struct foo f = { 0 }`, but that’ll only work properly if the first field is an integer. Maybe one could invent a nice preprocessor macro that is standard, nice _and_ convenient for this purpose?

**Comment by Talhada** [comment]
Hi, yes good point! Indeed it is a convenient way of initializing structures. To be Hines I've never seen it used in C before. Do you know how to make it work with msvc? The macro solution may be good as well. Thank you for the comments!

**Comment by frno7** [comment]
I haven’t found a nice proposal for a zero initialisation method yet, but we _might_ add the [`-pedantic`](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-pedantic-1) option to GCC to have it be strict, and

> issue all the warnings demanded by strict ISO C and ISO C++; reject all programs that use forbidden extensions, and some other programs that do not follow ISO C and ISO C++.

In practice, though, it’s probably a good compromise if GCC, Clang and MSVC are supported, as that may enable some extension that are worthwhile to improve the code.

**Comment by frno7** [comment]
Fixed with #7.
